src/controller/proxy/controller.go

```diff
@@ -175,35 +213,64 @@ func (c *controller) UseLocalManifest(ctx context.Context, art lib.ArtifactInfo,
 		return false, nil, errors.NotFoundError(fmt.Errorf("repo %v, tag %v not found", art.Repository, art.Tag))
 	}
 
+	// Use digest to retrieve manifest, digest is more stable than tag, because tag could be updated
```

Comment:
what you changed in this part is already covered previous line 202

Reply:
Previous line 202 is the return of a manifest list if it could be fetched from the cache. 
`return true, &ManifestList{content, string(desc.Digest), contentType}, nil`

But since manifests are not cached this returns a `not found error` in previous line 190 for GET .../manifests/{tag} requests referencing a tag.
`log.Debugf("Digest is not found in manifest list cache, key=cache:%v", manifestListKey(art.Repository, art))`

My intention with this part in line 223-234 was to check prior to the manifest list cache if a manifest with the digest already exists in the local repository and if found return it.

Reply:
Sorry I'm new to harbor and I think my proposed implementation would skip other middlewares, which might be a problem.